scipy/stats/_unuran/unuran_wrapper.pyx.templ

```diff
@@ -1370,6 +1401,510 @@ cdef class NumericalInversePolynomial(Method):
         return UError(max_error, mae)
 
 
+cdef class NumericalInverseHermite(Method):
+    """
+    NumericalInverseHermite(dist, *, domain=None, order=3, u_resolution=1e-12, construction_points=None, max_intervals=100000, guide_factor=1, random_state=None)
+
+    Hermite interpolation based INVersion of CDF (PINV).
+
+    HINV is a variant of numerical inversion, where the inverse CDF is approximated using
+    Hermite interpolation, i.e., the interval [0,1] is split into several intervals and
+    in each interval the inverse CDF is approximated by polynomials constructed by means
+    of values of the CDF and PDF at interval boundaries. This makes it possible to improve
+    the accuracy by splitting a particular interval without recomputations in unaffected
+    intervals. Three types of splines are implemented: linear, cubic, and quintic
+    interpolation. For linear interpolation only the CDF is required. Cubic interpolation
+    also requires PDF and quintic interpolation PDF and its derivative.
+
+    These splines have to be computed in a setup step. However, it only works for
+    distributions with bounded domain; for distributions with unbounded domain the tails
+    are chopped off such that the probability for the tail regions is small compared to
+    the given u-resolution.
+
+    The method is not exact, as it only produces random variates of the approximated
+    distribution. Nevertheless, the maximal numerical error in "u-direction" (i.e.
+    ``|U - CDF(X)|`` where ``X`` is the approximate percentile corressponding to the
+    quantile ``U`` i.e. ``X = approx_ppf(U)``) can be set to the
+    required resolution (within machine precision). Notice that very small values of
+    the u-resolution are possible but may increase the cost for the setup step.
+
+    Parameters
+    ----------
+    dist : object
+        An instance of a class with a ``cdf`` and optionally a ``pdf`` and ``dpdf`` method.
+
+        * ``cdf``: CDF of the distribution. The signature of the CDF is expected to be:
+          ``def cdf(self, x: float) -> float``. i.e. the CDF should accept a Python
+          float and return a Python float.
+        * ``pdf``: PDF of the distribution. This method is optional when ``order=1``.
+          Must have the same signature as the PDF.
+        * ``dpdf``: Derivative of the PDF w.r.t the variate (i.e. ``x``). This method is
+          optional with ``order=1`` or ``order=3``. Must have the same signature as the CDF.
+
+    domain : list or tuple of length 2, optional
+        The support of the distribution.
+        Default is ``None``. When ``None``:
+
+        * If a ``support`` method is provided by the distribution object
+          `dist`, it is used to set the domain of the distribution.
+        * Otherwise the support is assumed to be :math:`(-\infty, \infty)`.
+
+    order : int, optional
```

Comment:
What is the default? Consider giving the set with the default listed first rather than specifying `int`.

Reply:
Done.